lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -124,12 +124,11 @@ FrontendInputsAndOutputs::getRequiredUniquePrimaryInput() const {
 }
 
 std::string FrontendInputsAndOutputs::getStatsFileMangledInputName() const {
-  // FIXME: "batch" should probably be some concatenation of all the primary
-  // input names, in order to keep the stats file names unique. (Or perhaps just
-  // the first primary?)
+  // Use the first primary, even if there are multiple primaries.
+  // That's enough to keep the file names unique.
   return isWholeModule()
              ? "all"
-             : primaryInputCount() == 1 ? firstPrimaryInput().file() : "batch";
+             : hasPrimaryInputs() ? firstPrimaryInput().file() : "batch";
```

Comment:
We think the else branch is dead now, right?

Reply:
Thanks! I've fixed it.